tests/unit/restapi/v1/test_queue.py

```diff
@@ -0,0 +1,500 @@
+
+
+def assert_cannot_rename_queue_with_existing_name(
+    client: FlaskClient,
+    queue_id: int,
+    existing_name: str,
+    existing_description: str,
+) -> None:
+    """Assert that renaming a queue with an existing name fails.
+    Args:
+        client: The Flask test client.
+        queue_id: The id of the queue to rename.
+        name: The name of an existing queue.
+    Raises:
+        AssertionError: If the response status code is not 400.
+    """
+    response = modify_queue(
+        client=client,
+        queue_id=queue_id,
+        new_name=existing_name,
+        new_description=existing_description,
+    )
+    assert response.status_code == 400
+
+
+# -- Tests -----------------------------------------------------------------------------
+
+
+@pytest.mark.v1
+def test_create_queue(
+    client: FlaskClient,
+    db: SQLAlchemy,
+    auth_account: dict[str, Any],
+) -> None:
+    """Test that queues can be correctly registered and retrieved using the API.
+
+    Given an authenticated user, this test validates the following sequence of actions:
+
+    - The user registers a queue named "tensorflow_cpu".
+    - The response is valid matches the expected values given the registration request.
+    - The user is able to retrieve information about the queue using the queue id.
+    """
+    name = "tensorflow_cpu"
+    description = "The first queue."
+    user_id = auth_account["user_id"]
+    group_id = auth_account["default_group_id"]
+    queue1_response = actions.register_queue(
+        client, name=name, description=description, group_id=group_id
+    )
+    queue1_expected = queue1_response.get_json()
+    assert_queue_response_contents_matches_expectations(
+        response=queue1_expected,
+        expected_contents={
+            "name": name,
+            "description": description,
+            "user_id": user_id,
+            "group_id": group_id,
+        },
+    )
+    assert_retrieving_queue_by_id_works(
+        client, queue_id=queue1_expected["queueId"], expected=queue1_expected
+    )
+
+
+@pytest.mark.v1
+def test_queue_get_all(
+    client: FlaskClient,
+    db: SQLAlchemy,
+    auth_account: dict[str, Any],
+    registered_queues: dict[str, Any],
+) -> None:
+    """Test that all queues can be retrieved.
+
+    Given an authenticated user and registered queues, this test validates the following
+        sequence of actions:
+
+    - A user registers three queues, "tensorflow_cpu", "tensorflow_gpu", "pytorch_cpu".
+    - The user is able to retrieve a list of all registered queues.
+    - The returned list of queues matches the full list of registered queues.
+    """
+    queue1_expected = registered_queues["queue1"]
+    queue2_expected = registered_queues["queue2"]
+    queue3_expected = registered_queues["queue3"]
+    queue_expected_list = [queue1_expected, queue2_expected, queue3_expected]
+
+    assert_retrieving_queues_works(client, expected=queue_expected_list)
```

Comment:
Double check that the behavior in `assert_retrieving_queues_works` is consistent with the objects in `queue_expected_list`. To me, it looks like `registered_queues` contains `TestResponse` objects (no `.get_json()` is issued in that fixture), but the assert function looks like its treating these objects like dictionaries.

Reply:
I am making all of the actions return `TestResponse` (this is consistent with actions in `test_queue.py`) and fixtures return a dictionary. This required moving the logic for adding additional information to the reponse in `actions.register_user` to the `registered_users` fixture.

This allows the create/register tests validate the response codes when calling actions directly, but other tests that utilize the fixtures (where the creation of the resource is a given and not being tested) direct access to the underlying contents.